app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayDatasetActivity.kt

```diff
@@ -4,58 +4,73 @@ import android.app.Activity
-    private val spoon = DummyCategory("Spoon","Spoon", null)
-
-    private val dummyPresenter = DummyUIPresenter(DummyDatabaseService())
-
-    private val datasetImagesList = ArrayList<CategorizedPicture>()
+    private val staticDBManagement = DummyDatabaseManagement.staticDummyDatabaseManagement
+    private lateinit var categories: Set<Category>
+    private lateinit var datasetId: String
+    private lateinit var dataset: Dataset
+
+    private val addPictureContractRegistration =
+        registerForActivityResult(AddPictureActivity.AddPictureContract) { resultPair ->
+            if (resultPair == null) {
+                Toast.makeText(
+                    this,
+                    "The picture has not been saved in the dataset",
+                    Toast.LENGTH_SHORT
+                ).show()
+            } else {
+                val category = resultPair.first
+                val pictureUri = resultPair.second
+                lifecycleScope.launch {
+                    staticDBManagement.putPicture(pictureUri, category)
+                }
+            }
+        }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_display_dataset)
+
+        checkIntentExtras(intent.extras)
+
+        var representativePictures = setOf<CategorizedPicture>()
+
         lifecycleScope.launch {
-            datasetImagesList.add(dummyPresenter.getPicture(fork.name)!!)
-            datasetImagesList.add(dummyPresenter.getPicture(knife.name)!!)
-            datasetImagesList.add(dummyPresenter.getPicture(spoon.name)!!)
+            dataset = staticDBManagement.getDatasetById(datasetId)!!
+            categories = staticDBManagement.getCategories()
+            for (cat in categories) {
+                representativePictures =
+                    representativePictures.plus(staticDBManagement.getPicture(cat)!!)
+            }
+        }
 
-            val displayDatasetAdapter =
-                DisplayDatasetAdapter(datasetImagesList, this@DisplayDatasetActivity)
+        val displayDatasetAdapter =
+            DisplayDatasetAdapter(representativePictures, this@DisplayDatasetActivity)
 
-            findViewById<GridView>(R.id.display_dataset_imagesGridView).adapter =
-                displayDatasetAdapter
+        findViewById<GridView>(R.id.display_dataset_imagesGridView).adapter =
+            displayDatasetAdapter
 
-            findViewById<GridView>(R.id.display_dataset_imagesGridView).setOnItemClickListener { adapterView, view, i, l ->
-                val intent =
-                    Intent(this@DisplayDatasetActivity, DisplayImageSetActivity::class.java)
-                //TODO: All the images that belong to the specified category will be sent to DisplayImageSetActivity
-                intent.putExtra("display_image_set_images", (datasetImagesList[i]) as Parcelable)
-                startActivity(intent)
+        gridViewItemListener()
```

Comment:
This name isn't descriptive of what the function does imo.

```suggestion
        setGridViewItemListener()
```